tw-tkms-starter/src/main/java/com/transferwise/kafka/tkms/TkmsInterrupterService.java

```diff
@@ -0,0 +1,40 @@
+package com.transferwise.kafka.tkms;
+
+import com.transferwise.common.baseutils.concurrency.IExecutorServicesProvider;
+import com.transferwise.common.baseutils.concurrency.ScheduledTaskExecutor;
+import com.transferwise.common.baseutils.concurrency.ScheduledTaskExecutor.TaskHandle;
+import java.time.Duration;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Slf4j
+public class TkmsInterrupterService implements ITkmsInterrupterService, InitializingBean {
+
+  @Autowired
+  private IExecutorServicesProvider executorServicesProvider;
+  private ScheduledTaskExecutor scheduledTaskExecutor;
+
+  public void afterPropertiesSet() {
```

Comment:
Can `ScheduledTaskExecutor` be already passed from the constructor as we should have access to it in `TkmsConfiguration`?

Reply:
Currently no, currently only the IExecutorServicesProvider bean is available.